src/tools/ts-generator/main.cpp

```diff
@@ -92,6 +97,35 @@ std::vector<Data::ThermalCluster*> getClustersToGen(Data::AreaList& areas,
     return clusters;
 }
 
+std::vector<Data::AreaLink*> getLinksToGen(Data::AreaList& areas,
+                                           const std::string& clustersToGen)
+{
+    std::vector<Data::AreaLink*> links;
+    const auto ids = splitStringIntoPairs(clustersToGen, ';', '.');
+
+    for (const auto& [areaID, linkID] : ids)
+    {
+        logs.info() << "Generating ts for area: " << areaID << " and cluster: " << linkID;
+
+        auto* area = areas.find(areaID);
+        if (!area)
+        {
+            logs.warning() << "Area not found: " << areaID;
+            continue;
+        }
+
+        auto it = std::ranges::find_if(area->links, [&linkID](auto& l)
+                { return l.second->with->id == linkID;});
+
+        if (it != area->links.end())
+            links.push_back(it->second);
+        else
+            logs.warning() << "Link not found: " << linkID;
+    }
+
+    return links;
+}
+
 int main(int argc, char *argv[])
 {
     TsGeneratorSettings settings;
```

Comment:
Indeed this could/should be improved, and is one of the reason that we made a specific tool for it.
However it's way more complex to load individual things rather than the whole study, and may be done in the future if users want a faster generator.

Reply:
ok